clients/drcachesim/tracer/tracer.cpp

```diff
@@ -856,6 +850,27 @@ create_buffer(per_thread_t *data)
     }
 }
 
+static size_t
+get_v2p_buffer_size()
+{
+    // The handoff interface requites we use dr_raw_mem_alloc and thus page alignment.
+    // We use one page which is 256 entries for 4K pages which is enough for
+    // most cases.
+    return dr_page_size();
+}
+
+static void
+create_v2p_buffer(per_thread_t *data)
+{
+    // The handoff interface requites we use dr_raw_mem_alloc and thus page alignment.
+    data->v2p_buf = (byte *)dr_raw_mem_alloc(get_v2p_buffer_size(),
```

Comment:
For apps with a lot of threads, can this increased memory usage cause issues?

Reply:
Yes, esp on an ARM machine with 64K pages: with 10K threads, that's 640MB.  Expanded the comment to talk about this.  We could use a smaller buffer when the handoff interface is not in use, or change the interface to use a different free function: I put that in a comment.